### PR TITLE
Implement button and use new tailwind styles

### PR DESCRIPTION
**Pull request by DexterHill0**
I have modified the `Button` and styled it to look like the GovUK component. This includes all states `hover`, `active`, `focus`, etc. I have created new stories and tests for this new button.

Due to the changes in the tailwind config and the names of colours and other properties, I have updated the existing components and tests to use these new names.

I have added `tailwind-merge` as a dependency, so components can optionally take extra user supplied class names and they will be applied correct. `react-icons` v5.4.0 has been added as a peer dependency.

To explain the components and some of the design decisions I have updated the readme as well.

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "recommendations": ["bradlc.vscode-tailwindcss"]
```
Comment (marksmall) [inline .vscode/extensions.json:2]: We will want to add more to this, but this is fine to start with, at least we will want:

- dbaeumer.vscode-eslint
- esbenp.prettier-vscode

There will definitely be more, but these are probably the minimum.

File: package.json [hunk 2]
```diff
@@ -109,9 +111,13 @@
...
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-icons": "^5.4.0",
     "tailwindcss": "^3.4.1"
   },
   "release": {
     "extends": "./release.config.js"
+  },
+  "dependencies": {
+    "tailwind-merge": "2.6.0"
```
Comment (marksmall) [inline package.json:121]: You've made this a dependency, but `react-icons` a peer dependency, I'd expect these to be the same.

File: src/components/Button/Button.stories.tsx [hunk 3]
```diff
@@ -1,28 +1,36 @@
...
+          Inverse
+        </Button>
+      </div>
+      <div>
+        <Button type="primary" disabled>
+          Disabled Button
+        </Button>
+      </div>
+    </div>
+  ),
```
Comment (marksmall) [inline src/components/Button/Button.stories.tsx:35]: It's hard to say if all buttons should be on one page or many, but I'm fine with this.

File: src/components/Button/Button.test.tsx [hunk 4]
```diff
@@ -4,34 +4,43 @@ import { describe, expect, it } from 'vitest';
...
-    expect(button).toHaveClass('bg-govukGreen');
-    expect(button).toHaveTextContent('Click me');
+    expect(button).toHaveClass('bg-white');
+    expect(button).toHaveTextContent('Inverse');
   });
-
-  it('renders with correct children content', () => {
-    render(<Button color="green">Submit</Button>);
-
+  it('renders with disabled style', () => {
```
Comment (marksmall) [inline src/components/Button/Button.test.tsx:25]: More for future, but can you separate out your tests by an empty line, it really helps with git interactive adds, as well as readability etc.

**Review by marksmall — APPROVED** [review]
few comments but happy to approve this, leave it up to you as to whether to merge, or you want to make changes.
